### PR TITLE
Add JMX contract tests

### PR DESCRIPTION
**Pull request by jefchien**
*Description of changes:* Adds contract tests to verify the expected behavior of the following changes:
- https://github.com/aws-observability/aws-otel-java-instrumentation/pull/817
- https://github.com/aws-observability/aws-otel-java-instrumentation/pull/898
- https://github.com/aws-observability/aws-otel-java-instrumentation/pull/901

Starts up sample apps with the instrumented SDK and verifies that metrics are received for each of the supported JMX target systems.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: appsignals-tests/images/kafka/kafka-consumers/src/main/java/App.java
Comment (bjrara) [inline appsignals-tests/images/kafka/kafka-consumers/src/main/java/App.java:1]: Can you help explain why we need to make changes in the kafka application?

Comment (jefchien) [reply appsignals-tests/images/kafka/kafka-consumers/src/main/java/App.java:1]: When the KafkaConsumer is created in the anonymous function, it isn't around long enough for the JMX metrics to get registered and recorded. Moving it to initialize on startup allows the Mbeans to be available to scrape.

Comment (bjrara) [reply appsignals-tests/images/kafka/kafka-consumers/src/main/java/App.java:1]: Can we add a comment?

File: appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/utils/MockCollectorClient.java [hunk 1]
```diff
@@ -150,11 +160,24 @@ public List<ResourceScopeMetric> getMetrics(Set<String> presentMetrics) {
...
+              Set<String> diff = Sets.difference(presentMetrics, receivedMetrics);
+              if (!diff.isEmpty()) {
+                logger.info(
+                    "received metrics diff {} vs {}: {}",
+                    presentMetrics.size(),
+                    receivedMetrics.size(),
+                    diff);
+              }
+              boolean isReady;
+              if (exactMatch) {
```
[the remark this answers is not in the document]
Comment (jefchien) [reply appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/utils/MockCollectorClient.java]: Ended up just reverting this file and reusing the `getRuntimeMetrics` function.

File: appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/misc/jmx/KafkaBrokerMetricsTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  @Override
+  protected String getApplicationWaitPattern() {
+    return ".* Kafka Server started .*";
+  }
+
+  @Override
+  protected Map<String, String> getApplicationEnvironmentVariables() {
+    return Map.of(
+        "KAFKA_OPTS", "-javaagent:" + MOUNT_PATH,
```
[the remark this answers is not in the document]
Comment (jefchien) [reply appsignals-tests/contract-tests/src/test/java/software/amazon/opentelemetry/appsignals/test/misc/jmx/KafkaBrokerMetricsTest.java]: Yeah, it seems like the kafka broker container will not start. Moved it to the additional env vars and added comments.

**Review by bjrara — APPROVED** [review]

**Pull request merged**